netbox/client.go

```diff
@@ -68,25 +78,40 @@ func NewClient(addr string, client *http.Client) (*Client, error) {
 // If a nil Valuer is specified, no query parameters will be sent with the
 // request.
 func (c *Client) NewRequest(method string, endpoint string, options Valuer) (*http.Request, error) {
-	rel, err := url.Parse(endpoint)
-	if err != nil {
-		return nil, err
-	}
+	return c.NewDataRequest(method, endpoint, options, nil)
+}
 
+// NewDataRequest creates a HTTP request using the input HTTP method, URL
+// endpoint, a Valuer which creates URL parameters for the request, and
+// a io.Reader as the body of the request.
+//
+// If a nil Valuer is specified, no query parameters will be sent with the
+// request.
+// If a nil io.Reader, no body will be sent with the request.
```

Comment:
Add another newline before this to make the documentation more readable in godoc.

Add: "If a nil io.Reader *is specified*, ..."